src/Language/PureScript/Docs/Prim.hs

```diff
@@ -0,0 +1,205 @@
+  , "    z = 1.224e6 :: Number"
+  ]
+
+int :: Declaration
+int = primType "Int" $ unlines
+  [ "A 32-bit signed integer. See the purescript-integers package for details"
+  , "of how this is accomplished when compiling to JavaScript."
+  , ""
+  , "Construct values of this type with literals:"
+  , ""
+  , "    x = 23 :: Int"
+  ]
+
+string :: Declaration
+string = primType "String" $ unlines
+  [ "A String. As in JavaScript, String values represent sequences of UTF-16"
+  , "code units, which are not required to form a valid encoding of Unicode"
+  , "text (for example, lone surrogates are permitted)."
+  , ""
+  , "Construct values of this type with literals, using double quotes `\"`:"
+  , ""
+  , "    x = \"hello, world\" :: String"
+  , ""
+  , "Multi-line string literals are also supported with triple quotes (`\"\"\"`)."
+  ]
+
+char :: Declaration
+char = primType "Char" $ unlines
+   [ "A single character (UTF-16 code unit). The JavaScript representation is a"
+   , "normal String, which is guaranteed to contain one code unit. This means"
+   , "that astral plane characters (i.e. those with code point values greater"
+   , "than 0xFFFF) cannot be represented as Char values."
+   , ""
+   , "Construct values of this type with literals, using single quotes `'`:"
+   , ""
+   , "    x = '0' :: Char"
+   ]
+
+boolean :: Declaration
+boolean = primType "Boolean" $ unlines
+  [ "A JavaScript Boolean value."
+  , ""
+  , "Construct values of this type with the literals `true` and `false`."
+  ]
+
+partial :: Declaration
+partial = primClass "Partial" $ unlines
+  [ "The Partial type class is used to indicate that a function is *partial,*"
+  , "that is, it will throw an error for some inputs. For more information,"
+  , "see [the Partial type class guide](https://github.com/purescript/documentation/blob/master/guides/The-Partial-type-class.md)."
+  ]
+
+fail :: Declaration
+fail = primClass "Fail" $ unlines
+  [ "The Fail type class is part of the custom type errors feature. To provide"
+  , "a custom type error when someone tries to use a particular instance,"
+  , "write that instance out with a Fail constraint."
+  , ""
+  , "For more information, see"
+  , "[the Custom Type Errors guide](https://github.com/paf31/24-days-of-purescript-2016/blob/master/21.markdown)."
+  ]
+
+typeConcat :: Declaration
+typeConcat = primType "TypeConcat" $ unlines
+  [ "The TypeConcat type constructor concatenates two Symbols in a custom type"
+  , "error."
+  , ""
+  , "For more information, see"
+  , "[the Custom Type Errors guide](https://github.com/paf31/24-days-of-purescript-2016/blob/master/21.markdown)."
+  ]
+
+typeString :: Declaration
+typeString = primType "TypeString" $ unlines
+  [ "The TypeString type constructor renders any concrete type into a Symbol"
+  , "in a custom type error."
+  , ""
+  , "For more information, see"
+  , "[the Custom Type Errors guide](https://github.com/paf31/24-days-of-purescript-2016/blob/master/21.markdown)."
```

Comment:
I wonder if we should just create a placeholder page in the docs repo for now instead of linking to "24 Days" which is more likely to become out of date.